sway/desktop/xdg_shell.c

```diff
@@ -231,16 +231,20 @@ static void _close(struct sway_view *view) {
 		return;
 	}
 	struct wlr_xdg_surface *surface = view->wlr_xdg_surface;
-	if (surface->role == WLR_XDG_SURFACE_ROLE_TOPLEVEL) {
-		wlr_xdg_surface_send_close(surface);
+	if (surface->role == WLR_XDG_SURFACE_ROLE_TOPLEVEL
+		&& surface->toplevel) {
```

Comment:
Style: we use two tabs when indenting multi-line conditions. Can you add one more?